src/main.ml

```diff
@@ -146,7 +146,9 @@ let main port certfile ciphersuites =
                (fun () ->
                   Lwt.finalize
                     (fun () -> (
-                      inc_conn () >>=
+                      inc_conn () >>= fun () -> (
+                        Lwt.return (Lwt_unix.setsockopt fd Lwt_unix.SO_KEEPALIVE true)
+                      ) >>=
```

Comment:
This can be simplified to:
```ocaml
inc_conn () >>= fun () ->
Lwt_unix.setsockopt fd Lwt_unix.SO_KEEPALIVE true;
xapi_says_use_tls >>= fun tls ->
let tls_role = if tls then tls_server_role else None in
handle_connection fd tls_role)
```
Here we're using the idiomatic `Lwt` formatting: we don't indent the lines after the `bind`.

Reply:
No, because `xapi_says_use_tls` takes `unit`, so that code gives
```
File "src/main.ml", line 151, characters 22-39:
Error: This expression has type unit -> bool Lwt.t
       but an expression was expected of type 'a Lwt.t
```
referring to characters `xapi_says_use_tls`.

But we can fix this be changing the `;` in your code snippet to `|>` so I'll do that.

Reply:
Or maybe it would be clearer to have
```
...;
xapi_says_use_tls () >>=
```
rather than
```
... |>
xapi_says_use_tls >>=
```

Reply:
Oops, I've update my comment - I think it's much clearer if we explicitly pass the `()` to it.